components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs

```diff
@@ -1600,14 +1602,16 @@ public sealed class GatewayServer : SessionServerBase, IServer {
                 ApplicationName = "Opc UA Gateway Server",
                 ApplicationType = Opc.Ua.ApplicationType.ClientAndServer,
                 ApplicationUri =
-                    $"urn:{Utils.GetHostName()}:Microsoft:OpcGatewayServer",
+                    $"urn:{Dns.GetHostName()}:Microsoft:OpcGatewayServer",
                 ProductUri = "http://opcfoundation.org/UA/SampleServer",
 
                 SecurityConfiguration = new SecurityConfiguration {
                     ApplicationCertificate = new CertificateIdentifier {
-                        StoreType = "Directory",
-                        StorePath =
-                "OPC Foundation/CertificateStores/MachineDefault",
+                        StoreType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                                "X509Store" : "Directory",
+                        StorePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                                "CurrentUser\\UA_MachineDefault" :
+                                "OPC Foundation/CertificateStores/MachineDefault",
```

Comment:
>MachineDefaul [](start = 66, length = 13)

pki/own

Reply:
right, I'll change the structure here as well